content/articles/multi-label-classification-with-scikit-multilearn/index.md

```diff
@@ -0,0 +1,654 @@
+
+For example, when predicting a given movie category, it may belong to horror, romance, adventure, action, or simultaneously. In this example, we have multi-labels that can be assigned to a given movie. In multi-class classification, an input belongs to only a single label. For example, when predicting if a given image belongs to a cat or a dog, the output can be either a cat or dog but not both at the same time.
+
+In this tutorial, we will be dealing with multi-label text classification, and we will build a model that classifies a given text input into different categories. Our text input can belong to multiple categories or labels at the same time.
+
+We will use scikit-multi learn in building our model. Scikit-multi learn is a python library built on top of scikit-learn and is best suited for multi-label classification.
+
+### Table of contents
+
+- [Prerequisites](#prerequisites)
+- [Introduction](#introduction)
+- [Problem transformation](#problem-transformation)
+- [Adapted algorithm](#adapted-algorithm)
+- [Ensemble methods](#ensemble-methods)
+- [Dataset](#dataset)
+- [Loading exploratory data analysis packages](#loading-exploratory-data-analysis-packages)
+- [Loading dataset](#loading-dataset)
+- [Checking data structure](#checking-data-structure)
+- [Datatype of our labels](#datatype-of-our-labels)
+- [Loading machine learning packages](#loading-machine-learning-packages)
+- [Installing Scikit-Multilearn](#installing-scikit-multilearn)
+- [Importing problem transformation packages](#importing-problem-transformation-packages)
+- [Text preprocessing](#text-preprocessing)
+- [Explore the dataset for noise](#explore-the-dataset-for-noise)
+- [Extracting stop words](#extracting-stop-words)
+- [Removing stop words](#removing-stop-words)
+- [Feature engineering](#feature-engineering)
+- [Dataset split](#dataset-split)
+- [Builing model](#builing-model)
+- [Binary relevance technique](#binary-relevance-technique)
+- [Classifier chains technique](#classifier-chains-technique)
+- [Labelpowerset technique](#labelpowerset-technique)
+- [Making a single prediction](#making-a-single-prediction)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites
+
+A reader must have:
+
+1. A good understanding of [Python](/engineering-education/python-projects-for-beginners/).
+2. A good understanding of [machine learning modeling](/engineering-education/house-price-prediction/).
+3. Download the dataset [here](https://drive.google.com/file/d/1_scJyWek8DJUZ14_wZ8GXZslpwT8ubzt/view?usp=sharing).
+4. Have some knowledge of [Pandas](https://numpy.org/) and [Numpy.](https://numpy.org/)
+   > NOTE: To follow along easily, use [Google Colab](https://research.google.com/) to build your model.
+
+### Introduction
+
+We have different methods and techniques used for multi-label classification. These methods are as follows.
```

Comment:
Please write a brief intro, maybe a paragraph or two, before diving into the different methods needed.